### PR TITLE
Feature/44143

### PR DESCRIPTION
**Pull request by anqaka**
dev-44143
Implement focus state in components

### CODE REVIEW

File: components/01-globals/_focus/_focus.scss [hunk 1]
```diff
@@ -0,0 +1,27 @@
+*:focus {
+    @include focus();
+}
+
+// // Safari 10.1+
```
Comment (Aekal) [inline components/01-globals/_focus/_focus.scss]: Should these lines be commented?

File: components/01-globals/typography/link/_link.scss [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+$link__color                         : $blue !default;
+$link__decoration                    : underline !default;
+$link__font-family                   : $font-family-base !default;
+$link__font-weight                   : $font-weight-normal !default;
+$link__color-hover                   : $color-secondary !default;
+$link__line-height                   : $font-line-height !default;
+$link__transition                    : $transition-base !default;
+$link__text-decoration               : none !default;
+$link__color--invert                 : $gray-darker !default;
+$link__color-hover--invert           : $blue !default;
```
Comment (Aekal) [inline components/01-globals/typography/link/_link.scss]: Could you change the alignment to match
```
$link__color              : $blue !default;
$link__decoration         : underline !default;
$link__font-family        : $font-family-base !default;
$link__font-weight        : $font-weight-normal !default;
$link__color-hover        : $color-secondary !default;
$link__line-height        : $font-line-height !default;
$link__transition         : $transition-base !default;
$link__text-decoration    : none !default;
$link__color--invert      : $gray-darker !default;
$link__color-hover--invert: $blue !default;
```
as we have in all files (I use Alignment 0.3.0 extension in vscode for that)?

File: components/01-globals/variables/_variables.scss [hunk 3]
```diff
@@ -1,87 +1,100 @@
...
 
 // Outline
-$outline-base               : none !default;
+$outline-base                    : none !default;
+
+// Focus inline
+$color-focus-inline              : $white !default;
+$color-focus-inline-dark        : $gray-darkest !default;
+$bg-focus-inline                 : $color-focus !default;
+$bg-focus-inline-dark           : $color-primary !default;
```
Comment (Aekal) [inline components/01-globals/variables/_variables.scss]: Please fix the alignment above

File: components/02-elements/dropdown-list/_dropdown-list.scss [hunk 4]
```diff
@@ -158,8 +158,8 @@ $dropdown-list__content-margin--screen-m  : $spacer $spacer--medium !default;
             font-weight: $dropdown-list__item-font-weight--inner;
             &:hover,
             &:focus {
-                color: $dropdown-list__item-color--inner;
-                text-decoration: none;
+                color: $dropdown-list__item-color-hover--inner;
+                // text-decoration: underline;
```
Comment (Aekal) [inline components/02-elements/dropdown-list/_dropdown-list.scss]: Please remove this line if it's not needed

File: components/02-elements/form/select/_select.scss [hunk 5]
```diff
@@ -64,6 +68,13 @@ $select__field-list-border-radius--single: 20px !default;
         }
     }
 
+    &.is-focused {
+        .select__field-list {
+            @include focus-input();
+        }
+
```
Comment (Aekal) [inline components/02-elements/form/select/_select.scss]: Empty line

File: components/02-elements/form/select/_select.scss [hunk 6]
```diff
@@ -21,6 +21,10 @@ $select__field-list-border-radius--single: 20px !default;
             border: $select__border;
             border-radius: $select__border-radius;
             appearance: none;
+
```
Comment (Aekal) [inline components/02-elements/form/select/_select.scss]: I think we can add `transition: $transition-base` here in `select`, `input` and `textarea` fields. I think it looks nicer with small animation :)

File: components/02-elements/list/list--icon.hbs [hunk 7]
```diff
@@ -1,7 +1,7 @@
 <{{{ listTag }}} class="list {{ class }}" {{{ attributes }}}>
     {{# each elements }}
         <{{{ ../elementTag }}} class="list__item {{ this.class }}">
-            <a href="{{ iconItemLink }}" class="list__icon-link {{this.linkClass}} {{{ linkAttributes }}}">
+            <a href="{{ iconItemLink }}" class="list__icon-link {{this.linkClass}}" {{{ linkAttributes }}}">
```
Comment (Aekal) [inline components/02-elements/list/list--icon.hbs]: Unnecessary `"` after `{{{ linkAttributes }}}`

**Comment by szafran89** [comment]
@anqaka Can you also adjust `alpaca theme` to keep the same version?

**Comment by anqaka** [comment]
@szafran89 yes, theme will be updated too: https://github.com/SnowdogApps/magento2-alpaca-theme/pull/125 I'll update the filters too if needed

**Comment by anqaka** [comment]
@Aekal I've updated dropdown list styles, I've check components & theme with keyboard and it looks ok.

**Pull request merged**